resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/internal/builders/RefreshLocalJavaFileBuilder.java

```diff
@@ -40,15 +40,9 @@ protected IProject[] build(int kind, Map<String, String> args, IProgressMonitor
 		IFile file = project.getFile("A.java");
 		IPath localLocation = project.getLocation().append(file.getName());
 		java.io.File localFile = localLocation.toFile();
-		try (FileOutputStream out = new FileOutputStream(localFile)) {
-			if (localFile.exists()) {
-				localFile.delete();
-			}
-
-			out.write("public class A {}".getBytes());
-		} catch (IOException streamCloseIgnored) {
-			// ignore;
-		} catch (Exception e) {
+		try {
+			Files.writeString(localFile.toPath(), "public class A {}");
```

Comment:
Nit: This will write always with UTF8 - the old solution wrote with the platform default locale. Both of these are bonkers, I assume.

We might want to make the charset explicit by using

```suggestion
			Files.writeString(localFile.toPath(), "public class A {}", Charset.forName(file.getCharset());
```

Reply:
The text written is ASCII only, it does not matter.